src/App.tsx

```diff
@@ -1,24 +1,209 @@
+import { Todo } from './types/Todo';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { Notifications } from './Components/Notifications';
+import { FilterType } from './types/FilterType';
+import { ErrorTypes } from './types/ErrorTypes';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [error, setError] = useState<ErrorTypes | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [processedTodos, setProcessedTodos] = useState<number[]>([]);
+
+  const visibleTodos = useMemo(() => (
+    getFilteredTodos(todos, filterType)
+  ), [todos, filterType]);
+
+  const activeTodos = useMemo(() => (
+    getFilteredTodos(todos, FilterType.ACTIVE)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    getFilteredTodos(todos, FilterType.COMPLETED)
+  ), [todos]);
+
+  const loadTodos = useCallback(async () => {
+    setError(null);
+
+    if (USER_ID) {
+      try {
+        const todoList = await getTodos(USER_ID);
+
+        setTodos(todoList);
+      } catch {
+        setError(ErrorTypes.LOAD);
+      }
+    }
+  }, [USER_ID]);
+
+  const handleAddTodo = async (newTitle: string) => {
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      if (!newTitle.trim()) {
+        setError(ErrorTypes.INPUT);
+        setIsLoading(false);
+
+        return;
+      }
+
+      if (USER_ID) {
+        setTempTodo({
+          id: 0,
+          userId: USER_ID,
+          title: newTitle,
+          completed: false,
+        });
+
+        const newTodo = await postTodo({
+          userId: USER_ID,
+          title: newTitle,
+          completed: false,
+        });
+
+        setTodos(prevTodo => [...prevTodo, newTodo]);
+      }
+    } catch {
+      setError(ErrorTypes.ADD);
+    } finally {
+      setIsLoading(false);
+      setTempTodo(null);
+    }
+  };
+
+  const handleUpdateTodo = useCallback(async (todo: Todo) => {
+    setError(null);
+    setProcessedTodos(removeTodos => [...removeTodos, todo.id]);
+
+    try {
+      const updatedTodo = { ...todo, completed: !todo.completed };
+
+      await updateTodo(todo.id, updatedTodo);
+    } catch {
+      setError(ErrorTypes.UPDATE);
+    } finally {
+      setIsLoading(false);
+      setError(null);
+      await loadTodos();
+      setProcessedTodos([]);
+    }
+  }, []);
+
+  const handleUpdateTitle = useCallback(
+    async (todo: Todo, newTitle: string) => {
+      setProcessedTodos(removeTodos => [...removeTodos, todo.id]);
+
+      try {
+        const updatedTitle = { ...todo, title: newTitle };
+
+        await updateTodo(todo.id, updatedTitle);
+      } catch {
+        setError(ErrorTypes.UPDATE);
+      } finally {
+        setError(null);
+        await loadTodos();
```

Comment:
you can just put in your state the returned todo from `await updateTodo(todo.id, updatedTitle);`
```
const updatedTodo = await updateTodo(todo.id, updatedTitle);
setState(prev => prev....reset todo with this updatedTodo.id)
```